src/pages/PromptPage.vue

```diff
@@ -92,6 +92,9 @@ async function updatePrompt() {
   } else if (router.currentRoute.value.params.slug) {
     prompt.value = promptStore.getPrompts.find((prompt) => prompt.slug === router.currentRoute.value.params.slug)
   }
+  if (prompt.value == undefined) {
+    router.push('/404')
+  }
```

Comment:
This can lead to an error. beacause We retrieve the value of prompt asynchronously.

The second reason is prompt default value is and empty object, and it's change to `undefine` only with the output off find function when it doesn't found the prompt. 

So we can have this only for `:slug` route.

Reply:
What are your thoughts? I ran a quick test and it appears that because the 'await' keyword is invoked  before the `promptStore` function call, the added if check won't run until the `promptStore` returns.
I threw together this test to see if it works as expected even though the await keywork is nested

`(async () => {
  let global;
  if(true){
    await new Promise((res, rej) => { // with 'await'
      //res('RESOLVED')
      rej("REJECTED")
    }).then((data)=>{
      global = data;  // data assigment in 'then block' also awaited.
    }).catch(err => {
      global = err
      console.log(err)
    })
  }
  console.log('[global variable: ]', global) // no error
})();`